BinaryObjectScanner.Protection/Macrovision.RipGuard.cs

```diff
@@ -35,6 +36,19 @@ internal string RipGuardCheckPortableExecutable(string file, PortableExecutable
             if (name?.Equals("rgasdev", StringComparison.OrdinalIgnoreCase) == true)
                 return "RipGuard";
 
+            if (!File.Exists(file))
+                return null;
+
+            FileInfo fi = new FileInfo(file);
+
+            // So far, every seemingly-randomly named EXE on RipGuard discs have a consistent hash.
+            if (fi.Length == 49_152)
+            {
+                string sha1 = GetFileSHA1(file);
+                if (sha1 == "6A7B8545800E0AB252773A8CD0A2185CA2497938")
+                    return "RipGuard";
+            }
```

Comment:
Invert the `if` statement and wrap this entire block in the `if` instead. That will help avoid issues in the future if/when other checks are added. Also, for safety, please wrap in a `try/catch`. See other protections for examples.